src/lib/provable/crypto/poseidon.ts

```diff
@@ -221,6 +222,19 @@ const TokenSymbolPure: ProvableExtended<
   check({ field }: TokenSymbol) {
     rangeCheckN(48, field);
   },
+  toValue({ symbol }) {
+    return symbol;
+  },
+  fromValue(symbol: string | TokenSymbol) {
+    if (typeof symbol !== 'string') return symbol;
+    let bytesLength = new TextEncoder().encode(symbol).length;
+    if (bytesLength > 6)
```

Comment:
If you think it's a better idea, can you put `6`  into https://github.com/o1-labs/o1js/blob/main/src/lib/mina/constants.ts? It's a minor change, but it could really help explain later why it's this value.